.circleci/config.yml

```diff
@@ -7,29 +7,37 @@ commands:
       - checkout
       - run:
           name: Version Information
-          command: rustc --version; cargo --version; rustup --version
-      - run:
-          name: Install Dependencies
-          command: |
-            sudo sh -c 'echo "deb http://deb.debian.org/debian stretch-backports main" > /etc/apt/sources.list.d/backports.list'
-            sudo apt-get update
-            sudo apt-get install -y protobuf-compiler/stretch-backports cmake golang curl
-            sudo apt-get clean
-            sudo rm -r /var/lib/apt/lists/*
-            rustup component add clippy rustfmt
+          command: rustc --version; cargo --version; rustup --version; sccache --version
       - run:
           name: Setup Env
           command: |
             echo 'export TAG=0.1.${CIRCLE_BUILD_NUM}' >> $BASH_ENV
             echo 'export IMAGE_NAME=myapp' >> $BASH_ENV
+            echo 'export CARGO_INCREMENTAL=0' >> $BASH_ENV
+  restore-sccache-cache:
+    steps:
+      - restore_cache:
+          name: Restore sccache Cache
+          key: sccache-cache-{{ arch }}-{{ .Environment.CIRCLE_JOB }}
+  save-sccache-cache:
+    steps:
+      - run:
+          name: Show sccache Stats
+          command: sccache -s
+      - save_cache:
+          name: Save sccache Cache
+          key: sccache-cache-{{ arch }}-{{ .Environment.CIRCLE_JOB }}-{{ epoch }}
+          paths:
+            - "~/.cache/sccache"
 
 jobs:
   build:
     docker:
-      - image: circleci/rust:stretch
+      - image: metajack/circleci:latest
```

Comment:
We should use an image from our ECR repo, for this we'll have to set ECR credentials in Circle (need to cc people with permissions).
Even without custom image, we can use sccache with Circe cache, let's just first commit the caching?

Reply:
Yes, I'll push the image to ECR repo tomorrow.

It takes 1:30 to build sccache. That is about enough along with save/restore to eliminate any gains from caching. It does not seem worthwhile to enable it on its own.